src/main.rs

```diff
@@ -106,9 +115,11 @@ async fn main() -> std::io::Result<()> {
     );
 
     let pool_ref = pool.clone();
+    let search_config_ref = search_config.clone();
```

Comment:
This seems annoyingly inefficient, but I'm not sure if there's any way of improving this.  It's only cloning it once per time it runs, so it shouldn't be too bad.  My only comment is that the naming is inaccurate:  the `_ref` implies that this is some sort of shared reference, such as an `Arc`, while this is just a cloned structure.  Something like `thread_search_config` may be more accurate.